### ISSUE TITLE
Some Accessibility Issues

### ISSUE THREAD

**Issue by StommePoes** [thm/uinames#93]
I'm planning a PR for most of these, but figured it's good to have a clear list of issues for it anyway.
I'm referencing the WCAG for most of these, since it's a W3C standard and something most countries use as the basis of their laws (if they have any) regarding accessibility.

1. We're missing <a href="https://www.w3.org/TR/2008/REC-WCAG20-20081211/#meaning-doc-lang-id">WCAG Guideline 3.1.1</a> by not allowing the website language to be programmatically determined... meaning we just need a lang attribute on the HTML tag. The website is in English. Additionally it might be cool to have appropriate lang tags on the h1 as non-English names are displayed (for correct pronunciation by speech synthesis) but that would be an extra, I feel.
I like <a href="http://adrianroselli.com/2015/01/on-use-of-lang-attribute.html">Roselli's page on all the why's</a> for all the reasons a simple lang attribute is awesome.

2. We're really missing <a href="https://www.w3.org/TR/2008/REC-WCAG20-20081211/#keyboard-operation">WCAG Guideline 2.1</a>.
I initially ran across this useful little tool from a Tweet. I am sighted. The first thing I saw was commands to use the keyboard, which I thought was awesome! Then I realised I couldn't use any of the buttons to do anything, leaving me stuck with US names of both genders, one at a time. Nothing on the page really works with keyboard except the main script listening for the `spacebar`, `c`, and `ctrl+c`.

   Now it turns out, there are a set of nonstandard (but probably very useful for repeat users) keyboard shortcuts. However, if you are a keyboard-only user (rather than a mouse-using but keyboard-preferring power user), you won't know these shortcuts exist, because you need the keyboard to be able to click the info button to be able to read about the shortcuts in the first place.

   The current shortcuts are number-based, meaning they have are unlikely to conflict with most assistive tech key commands, which is good. The one I did run into was hitting 1 to reach the first-level heading (&lt;h1&gt;), but since there aren't any other headings on the page currently, this one didn't seem too bad to me. A regular AT user might disagree with me, in which case I'd suggest changing the 1 to some other number.

2.a Also, while focus isn't actually prevented from landing on the buttons, we <a href="https://www.w3.org/TR/2008/REC-WCAG20-20081211/#navigation-mechanisms-focus-order">aren't moving in the visual order</a> which most people would expect, and even if we were, <a href="https://www.w3.org/TR/2008/REC-WCAG20-20081211/#navigation-mechanisms-focus-visible">we can't see it</a>.

   That second link also points to the opposite problem: because the "popups" are hidden by being pulled offscreen or having an opacity of 0, they're still (invisibly) tabbable. So if my only recourse to navigating the page is to see what URL appears in my browser's lower-lefthand side, I see a whole lot of github URLs and other crazy stuff, with no idea what those are or mean. Things we're keeping hidden visually should also not be tabbable with keyboard.

   Keyboard operation is also lost with the Region selector. Only the list items can be clicked, and since they're not focusable things, they can't be clicked with keyboard. Avoiding the autocomplete functionality entirely, one could just type in the full name of a region... but again only clicking on an available region from the list below is the only way to select it. Hitting enter does nothing, nor is there a dedicated "submit" button or similar.

   Unfortunately, <a href="https://www.w3.org/TR/2008/REC-WCAG20-20081211/#ensure-compat">the semantics are poor</a> (<a href="https://marcysutton.com/links-vs-buttons-in-modern-web-applications/">the links should really be buttons</a>). However, while playing around with the problem, I found I couldn't really improve this as much as I wanted. 

   Buttons make the most sense for what the round circles near the top of the page do, but because the main idea of the page is the pressing of **spacebar** to do the site's main function, letting buttons be buttons on *this* page would result in undesired behaviour most of the time. Buttons listen for both enter and spacebar to trigger a click. My future-PR has a compromise which probably nobody in the accessibility world would like, but this *is* open source...

   Since <a href="https://www.paciellogroup.com/blog/2010/11/using-the-html-title-attribute/">the title attribute isn't really good for naming things</a>, but is still useful for letting sighted mouse users know what the buttons do, in my future-PR I left the titles in but substituted hidden text, aria-labels, and CSS "content" for improving some of the other use cases of naming things.
   I used the CSS content property making button names visible on focus for keyboarders. If using speech to control your computer, you'd still be stuck with something like "Press Tab" at least the first time to go through the buttons, but perhaps after seeing the names once you could call them out directly by name instead.

2.b Even when sighted, "standard" keyboard use is accounted for, there's still a <a href="https://www.w3.org/TR/2008/REC-WCAG20-20081211/#content-structure-separation-sequence">strange sequence issue</a> as far as where the buttons visually are and where the name is. I figured there was some good reason for placing all the buttons and controls and things after the main name so in my future-PR I've only rearranged the button order a bit, but kept the name first.

\3. The initial command "Press Spacebar" and the accessory texts that appear under it have <a href="https://www.w3.org/TR/2008/REC-WCAG20-20081211/#visual-audio-contrast-contrast">waaaay insufficient contrast</a>. It's actually also still too low even when the text turns completely white. <a href="https://leaverou.github.io/contrast-ratio/#%23fff-on-%2339844d">#39844d</a> would be sufficient for the smaller text (you can get away with a lot more for the larger text). Also, Lea's tool is really handy :)
   However I don't feel PRs that really change how someone else's thing looks get accepted much, so I did not include any contrast or colour changes. Certainly the green I found which has sufficient contrast with white isn't the same frog green as the site currently has. Maybe the text  colour would be better adjusted instead. Even if my PR gets accepted, this "insufficient contrast" will remain an open issue.
